common/sgx/eeid.c

```diff
@@ -0,0 +1,499 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <openenclave/bits/eeid.h>
+#include <openenclave/bits/sgx/sgxtypes.h>
+#include <openenclave/internal/crypto/sha.h>
+#include <openenclave/internal/hexdump.h>
+#include <openenclave/internal/malloc.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include <openenclave/internal/types.h>
+#include <openenclave/internal/utils.h>
+
+#include "sgxmeasure.h"
+
+#ifdef OE_BUILD_ENCLAVE
+#include <openenclave/enclave.h>
+#include "../../enclave/crypto/key.h"
+#include "../../enclave/crypto/rsa.h"
+#else
+#include <openenclave/host.h>
+#include <openssl/opensslv.h>
+#include <openssl/rsa.h>
```

Comment:
Yes, verify the base-image signature. There may be other ways to do that, but at the moment it's mbedTLS+OpenSSL only. I also need hash state de/serializing, which I haven't added to the 
bcrypt interface yet, mainly because I don't know how bcrypt keeps that information in memory.

Reply:
BTW, help very welcome, I think it would take me quite a while to test this on Windows.

Reply:
From my comment below, I think it's fine if out of scope for this PR. This code should just not be compiled on Windows and EEID can be marked as supported on Linux only.

Reply:
If EEID is moved out of experimental phase then I would want it to work on Windows too, but for now this is fine.

Reply:
Oh yeah, of course we would want it to work on all platforms in the long run!